ui/protocol_test.go

```diff
@@ -0,0 +1,51 @@
+package ui
+
+import (
+	"bytes"
+	"github.com/algorandfoundation/hack-tui/internal"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/x/exp/teatest"
+	"testing"
+	"time"
+)
+
+func Test_ProtocolViewRender(t *testing.T) {
+	status := internal.StatusModel{
+		HeartBeat:   make(chan uint64),
+		LastRound:   0,
+		NeedsUpdate: true,
+		State:       "SYNCING",
+	}
+
+	// Create the Model
+	m := MakeProtocolViewModel(&status)
+
+	tm := teatest.NewTestModel(
+		t, m,
+		teatest.WithInitialTermSize(80, 40),
+	)
+
+	// Wait for prompt to exit
+	teatest.WaitFor(
+		t, tm.Output(),
+		func(bts []byte) bool {
+			return bytes.Contains(bts, []byte("[UPDATE AVAILABLE]"))
```

Comment:
Besides update available should we check that the other strings are rendering properly, like the network name, protocol voting, node versions, etc?

Reply:
Definitely not worth it. The test suite is too experimental to provide good results. Most projects don't even test their tea components.